src/components/UI/__tests__/cluster_dashboard_item.js

```diff
@@ -0,0 +1,60 @@
+import 'jest-dom/extend-expect';
+import { MemoryRouter } from 'react-router-dom';
+import { render } from '@testing-library/react';
+import { ThemeProvider } from 'emotion-theming';
+import ClusterDashboardItem from 'UI/cluster_dashboard_item';
+import React from 'react';
+import theme from 'styles/theme';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
+  render(<ClusterDashboardItem />, div);
+});
+
+it('has links to the cluster detail page', () => {
+  const href = '/organizations/acme/clusters/1234';
+  const { container, debug } = render(
+    <ThemeProvider theme={theme}>
+      <MemoryRouter initialEntries={['/']}>
+        <ClusterDashboardItem cluster={{ owner: 'acme', id: '1234' }} />
+      </MemoryRouter>
+    </ThemeProvider>
+  );
+
+  const links = container.querySelectorAll('a');
+
+  expect(links[0]).toHaveAttribute('href', href);
+  expect(links[1]).toHaveAttribute('href', href);
+});
+
```

Comment:
@oponder nitpick: I think that one step further to this test would be: `links lead the user to the cluster details page`  we would fire a click event and see if the router leads us to the `href` it is essentially the same test but with the benefit of also testing the router.

Reply:
I did it, but I think maybe we're testing implementation detail at this point? 

This test doesn't wanna make sure that the memory router works, or that click events on `a` elements behave as expected. (Since that is a browser feature which I'm pretty sure we can rely on, and don't need to test)

Reply:
Yes, you are now testing implementation details. You can easily avoid it by testing if any of the elements in the cluster details page is rendered.

Reply:
> This test doesn't wanna make sure that the memory router works, or that click events on `a` elements behave as expected. (Since that is a browser feature which I'm pretty sure we can rely on, and don't need to test)

Maybe the router is broken for whatever reason and `Link` components are not rendering an `<a>`.

Reply:
> Yes, you are now testing implementation details. You can easily avoid it by testing if any of the elements in the cluster details page is rendered.

but then I have to render the full app... i think like this is ok for now right?

Reply:
And there's another thing to keep in mind: our users are using our application __with__ the router, the store, etc. They don't care about it, but what I mean is that if we want to test Happa in a way that ressembles how our users are using it we should go for integration tests in most cases.